developing_features/guides/new_project.md

```diff
@@ -0,0 +1,44 @@
+# [Developing Features](../developing_features.md) / Guides / Setting Up a New Project
+
+
+#### Checklist
+
+###### For a Public project
+
+ - Generate a new gem with `bundle gem abc`
+ - Commit it, before changing anything, as `[skip] Initial commit from Bundler 1.16.1 (1m)` (giving the appropriate version of Bundler)
+ - [Create a new repo on GitHub](https://github.com/new) (e.g. `cph/abc`)
+ - [Enable branch protection](https://help.github.com/articles/configuring-protected-branches/) on `master` and turn on these settings:
+    - Require pull request reviews before merging
+    - Require status checks to pass before merging
+       - Require branches to be up to date before merging
+       - Select Travis CI - Pull Request
+       > You won't be able to do this until you've pushed commits and had one build run on Travis.
+
+###### For a Private project
+
+ - Generate a new Rails project with `rails new xyz`
+ - Commit it, before changing anything, as `[skip] Initial commit from Rails 5.2.2 (1m)` (giving the appropriate version of Rails)
+ - [Create a new repo on GitHub](https://github.com/new) (e.g. `cph/xyz`)
+ - [Create a new app on Errbit](https://errbit.cphepdev.com/apps/new) (You may have to log out and log back in as an Admin)
+     - Fill in **GitHub Repo** (`cph/xyz`) so that lines in the backtraces can be linked to source
+     - Under **Watchers**, add the non-admin users in Errbit
+     - Under **Notification Service**, pick **Webhook** and fill in the URL `https://houst.in/projects/xyz/hooks/exception_report`
+ - Create a new project in Houston [for the appropriate Team](https://houst.in/)
+     - Give it the same slug as its GitHub repo
+     - Set **Version Control** to **Git** and fill in the the URL `git@github.com:cph/xys.git`
+     - Set **Error Tracker** to **Errbit** and fill in the AppId (from the URL `https://errbit.cphepdev.com/apps/:app_id`)
+     - Enable these features:
+        - **Feedback** (i.e. https://houst.in/feedback/by_project/xyz)
+        - **Goals** (so the new project can be added to Roadmaps)
+        - **Releases** (so that you can create release notes for the project)
+ - [Create a new job in Jenkins](https://ci.cphepdev.com/view/all/newJob)
+     - Give it the same name as its slug in Houston
+     - Copy its configuration from an existing job (e.g. **lsb**)
+     - Change **Respository URL** to the URL `git@github.com:cph/xys.git`
+ - [Enable branch protection](https://help.github.com/articles/configuring-protected-branches/) on `master` and turn on these settings:
+    - Require pull request reviews before merging
+    - Require status checks to pass before merging
+       - Require branches to be up to date before merging
+       - Select jenkins
+       > You won't be able to do this until you've pushed commits and had one build run on Jenkins whose status Houston has reported to GitHub.
```

Comment:
In order to get the project to build on Jenkins:

- Create a new feature branch (e.g. **configure-jenkins**)
- In the new branch, create a `bin/cibuild` script and make it executable (`chmod a+x`)
- Update `cibuild` with the build steps using one of the existing projects as a guide
- Add the `minitest`, `minitest-reporters`, `minitest-reporters-turn_reporter`, `simplecov`, `simplecov-json` gems to the Gemfile
- Copy `config/keypair.jenkins.pem` from one of the existing projects
- Update `test/test_helper.rb` with the following:
```
if ENV["COVERAGE"] == "on"
  require "simplecov"
  require "simplecov-json"
  SimpleCov.formatter = SimpleCov::Formatter::JSONFormatter
  SimpleCov.start "rails"
end
.
.
.
require "minitest/autorun"
require "minitest/reporters"
require "minitest/reporters/turn_reporter"

if ENV["CI"] == "true"
  MiniTest::Reporters.use! [ Minitest::Reporters::TurnReporter.new, MiniTest::Reporters::JUnitReporter.new ]
else
  MiniTest::Reporters.use! Minitest::Reporters::TurnReporter.new
end
```
- Create a commit and push it to Github
- Manually build the project from Jenkins using the branch's SHA

Reply:
If you configure Houston correctly for the project, you shouldn't need the last step; Houston will take care of starting the Jenkins run automatically.